service_configuration_lib/spark_config.py

```diff
@@ -172,11 +174,8 @@ def assume_aws_role(
     return resp['Credentials']
 
 
-def _pick_random_port(app_name):
+def _pick_random_port(preferred_port: int = 0):
```

Comment:
we might as well fully annotate the types here :)
```suggestion
def _pick_random_port(preferred_port: int = 0) -> int:
```